### ISSUE TITLE
Unable to unassign licenses using "User license allocations" when assigned licenses > 100

### ISSUE THREAD

**Issue by ehuggett** [iomad/iomad#273]
# Steps to reproduce:
1. Create a company and create enough users and and assign enough courses (that require a license to access) so that Users \* courses >= 100 (for example, 11 users and 10 courses or 101 users and 1 course etc)
2. Create a license that covers all courses and allows for > 100 allocations
3. assign all courses to all staff members
4. The column showing the assigned licenses will now show "Too many users to show" "Please use the search"
5. use the search to get < 100 matches
6. attempt to unallocate an unused license from the results
# Why this happens

The problem starts at [/blocks/iomad_company_admin/company_license_users_form.php#L255](https://github.com/iomad/iomad/blob/IOMAD_28_STABLE/blocks/iomad_company_admin/company_license_users_form.php#L255). get_selected_users will return nothing if the search form _had_ to be used.

```
$userstounassign = $this->currentusers->get_selected_users();
```

Eventually we end up at [/blocks/iomad_company_admin/lib/user_selectors.php#L1044](https://github.com/iomad/iomad/blob/IOMAD_28_STABLE/blocks/iomad_company_admin/lib/user_selectors.php#L1044)

```
$groupedusers = $this->find_users('');
```

And this is the problem, [find_users](https://github.com/iomad/iomad/blob/IOMAD_28_STABLE/blocks/iomad_company_admin/lib/user_selectors.php#L918) will always return [too many users](https://github.com/iomad/iomad/blob/IOMAD_28_STABLE/blocks/iomad_company_admin/lib/user_selectors.php#L961) if there are > 100 assigned licenses as the search search criteria used is not being passed to it.
# Solution?

I have changed this to the following and it seems to work, but I have not even started to look for any problems this might cause (none in theory?)

```
$search = optional_param($this->name . '_searchtext', '', PARAM_RAW);
$groupedusers = $this->find_users($search);
```

**Comment by ehuggett** [comment]
@turf212 was anything done to address this in 3.0?

setting validatinguserids to true (even if only temporarily) is another potential solution

something along the lines of 

```
$tmp = $this->validatinguserids;
$this->validatinguserids = true;
$groupedusers = $this->find_users('');
$this->validatinguserids = $tmp;
```

(edit: warning, not even executed let alone tested!)

**Comment by turf212** [comment]
The only thing I'll change is the PARAM_RAW to PARAM_CLEAN on the first solution.  Other than that, merged into 3.0 and thank you :)
